lib/handlers/quote/injector.ts

```diff
@@ -94,6 +100,13 @@ export class QuoteInjector extends ApiInjector<ContainerInjected, ApiRInj, Quote
       requestId,
     });
 
+    // TODO: add multiple providers for cross chain support
+    const tokenListProvider = new CachingTokenListProvider(
+      requestBody.tokenInChainId,
+      DEFAULT_TOKEN_LIST,
+      new NodeJSCache(new NodeCache())
+    );
+
```

Comment:
can have a mapping between `chainId` to `tokenListProvider`s and put that in the container

Reply:
we can just use the existing tokenFetcher with a slight modification, changed to that